lib/bitex/kyc_profile.rb

```diff
@@ -1,66 +1,141 @@
+
+    # @!attribute last_name
+    #   @return [String] Last Name.
+    attr_accessor :last_name
+
+    # @!attribute personal_id_number
+    #   @return [String] Personal ID Number.
+    attr_accessor :personal_id_number
+
+    # @!attribute personal_id_issuer
+    #   @return [String] ISO country code for the issuer of this ID.
+    attr_accessor :personal_id_issuer
+
+    # @!attribute personal_id_type
+    #   @return [symbol] Type of ID. [:passport]
+    attr_accessor :personal_id_type
+
+    # @!attribute tax_id
+    #   @return [Intger] Tax ID.
+    attr_accessor :tax_id
+
+    # @!attribute birth_date
+    #   @return [Time] Birth date.
+    attr_accessor :birth_date
+
+    # @!attribute nationality
+    #   @return [String] Nationality.
+    attr_accessor :nationality
+
+    # @!attribute gender
+    #   @return [Symbol] Gender.
+    attr_accessor :gender
+
+    # @!attribute occupation
+    #   @return [String]
+    attr_accessor :occupation
+
+    # @!attribute home_address
+    #   @return [String] Home address.
+    attr_accessor :home_address
+
+    # @!attribute work_address
+    #   @return [String] Work address.
+    attr_accessor :work_address
+
+    # @!attribute phone_numbers
+    #   @return [String] Phone numbers.
+    attr_accessor :phone_numbers
+
+    # @!attribute legal_entity
+    #   @return [Boolean] Is legal entity.
+    attr_accessor :legal_entity
+
+    # @!attribute politically_exposed_person
+    #   @return [Boolean] Is politically exposed.
+    attr_accessor :politically_exposed_person
+
+    # @!attribute usage_tier
+    #   @return [Symbol] Requested usage tier.
+    attr_accessor :usage_tier
+
+    # @!attribute accepted_usage_tier
+    #   @return [Symbol] Current usage tier as accepted by our compliance officers.
+    attr_accessor :accepted_usage_tier
 
     # @visibility private
+    # rubocop:disable Metrics/AbcSize
     def self.from_json(json)
-      new.tap do |thing|
-        thing.id = json[0]
-        thing.first_name = json[1]
-        thing.last_name = json[2]
-        thing.personal_id_number = json[3]
-        thing.personal_id_issuer = json[4]
-        thing.personal_id_type = json[5]
-        thing.tax_id = json[6]
-        thing.birth_date = json[7] ? Time.at(json[7]) : nil
-        thing.nationality = json[8]
-        thing.gender = json[9]
-        thing.occupation = json[10]
-        thing.home_address = json[11]
-        thing.work_address = json[12]
-        thing.phone_numbers = json[13]
-        thing.legal_entity = json[14]
-        thing.politically_exposed_person = json[15]
-        thing.usage_tier = json[16]
-        thing.accepted_usage_tier = json[17]
+      new.tap do |profile|
+        profile.id = json[0]
+        profile.first_name = json[1]
+        profile.last_name = json[2]
+        profile.personal_id_number = json[3]
+        profile.personal_id_issuer = json[4]
+        profile.personal_id_type = json[5].to_sym
+        profile.tax_id = json[6].to_i
```

Comment:
This is not an integer, it's an alphanumeric string.

Reply:
ready